server/methods/saveUserProfile.js

```diff
@@ -9,108 +9,108 @@ import { twoFactorRequired } from '../../app/2fa/server/twoFactorRequired';
-		check(settings, Object);
-		check(customFields, Match.Maybe(Object));
+function saveUserProfile(settings, customFields) {
+	if (!rcSettings.get('Accounts_AllowUserProfileChange')) {
+		throw new Meteor.Error('error-not-allowed', 'Not allowed', {
+			method: 'saveUserProfile',
+		});
+	}
+
+	if (!this.userId) {
+		throw new Meteor.Error('error-invalid-user', 'Invalid user', {
+			method: 'saveUserProfile',
+		});
+	}
+
+	const user = Users.findOneById(this.userId);
+
+	if (settings.realname || settings.username) {
+		if (!saveUserIdentity(this.userId, {
+			_id: this.userId,
+			name: settings.realname,
+			username: settings.username,
+		})) {
+			throw new Meteor.Error('error-could-not-save-identity', 'Could not save user identity', { method: 'saveUserProfile' });
+		}
+	}
 
-		if (!rcSettings.get('Accounts_AllowUserProfileChange')) {
-			throw new Meteor.Error('error-not-allowed', 'Not allowed', {
+	if (settings.statusText || settings.statusText === '') {
+		Meteor.call('setUserStatus', null, settings.statusText);
+	}
+
+	if (settings.statusType) {
+		Meteor.call('setUserStatus', settings.statusType, null);
+	}
+
+	if (settings.bio != null) {
+		if (typeof settings.bio !== 'string' || settings.bio.length > 260) {
+			throw new Meteor.Error('error-invalid-field', 'bio', {
 				method: 'saveUserProfile',
 			});
 		}
+		Users.setBio(user._id, settings.bio.trim());
+	}
 
-		if (!this.userId) {
-			throw new Meteor.Error('error-invalid-user', 'Invalid user', {
+	if (settings.nickname != null) {
+		if (typeof settings.nickname !== 'string' || settings.nickname.length > 120) {
+			throw new Meteor.Error('error-invalid-field', 'nickname', {
 				method: 'saveUserProfile',
 			});
 		}
-
-		const user = Users.findOneById(this.userId);
-
-		if (settings.realname || settings.username) {
-			if (!saveUserIdentity(this.userId, {
-				_id: this.userId,
-				name: settings.realname,
-				username: settings.username,
-			})) {
-				throw new Meteor.Error('error-could-not-save-identity', 'Could not save user identity', { method: 'saveUserProfile' });
-			}
-		}
-
-		if (settings.statusText || settings.statusText === '') {
-			Meteor.call('setUserStatus', null, settings.statusText);
-		}
-
-		if (settings.statusType) {
-			Meteor.call('setUserStatus', settings.statusType, null);
-		}
-
-		if (settings.bio != null) {
-			if (typeof settings.bio !== 'string' || settings.bio.length > 260) {
-				throw new Meteor.Error('error-invalid-field', 'bio', {
+		Users.setNickname(user._id, settings.nickname.trim());
+	}
+
+	if (settings.email) {
+		Meteor.call('setEmail', settings.email);
+	}
+
+	const canChangePasswordForOAuth = rcSettings.get('Accounts_AllowPasswordChangeForOAuthUsers');
+	if (canChangePasswordForOAuth || user.services?.password) {
+		// Should be the last check to prevent error when trying to check password for users without password
+		if (settings.newPassword && rcSettings.get('Accounts_AllowPasswordChange') === true) {
```

Comment:
We can remove the `=== true` part of this

Reply:
yep, this is an old code that was just re-indented, but I agree :+1: